### PR TITLE
Fabo/add disconnected bar

### PR DESCRIPTION
**Pull request by faboweb**
Closes #ISSUE

image

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by colw** [comment]
This is probably related to the underlying connection module, but what of the use case that you are connected, and then lose the connection?

I can disconnect in chrome, and get notification warnings (failed to fetch etc.) but the `DisconnectedBar` would not appear.

File: src/components/common/DisconnectedBar.vue [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<template>
+  <div v-if="connection.stopConnecting" class="disconnected-bar">
+    <i></i>
+    <p>
+      You are currently not connected to a node. Retry?
```
Comment (colw) [inline src/components/common/DisconnectedBar.vue]: I think the word 'Retry?' should initiate `reconnect` too.

File: src/vuex/getters.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ export const depositDenom = getters =>
     : `uatom`
 
 // connection
-export const approvalRequired = state => state.connection.approvalRequired
+export const connection = state => state.connection
```
Comment (colw) [inline src/vuex/getters.js]: Thinking about this. I believe we might be making too much use of getters, where we could just use `mapState` in the component.

Comment (faboweb) [reply src/vuex/getters.js]: That's a great idea. Hadn't considered this.

**Comment by faboweb** [comment]
> This is probably related to the underlying connection module, but what of the use case that you are connected, and then lose the connection?

That should be handled by this. After 5 attempts to connect to the websocket of the fullnode the bar appears.

**Comment by jbibla** [comment]
> I can disconnect in chrome, and get notification warnings (failed to fetch etc.) but the DisconnectedBar would not appear.

has this been resolved?

File: src/components/common/DisconnectedBar.vue [hunk 3]
```diff
@@ -2,8 +2,8 @@
   <div v-if="stopConnecting" class="disconnected-bar">
     <i></i>
     <p>
-      You are currently not connected to a node.
-      <a class="link" @click="reconnect">Retry</a>?
+      You are currently not connected.
+      <a class="link" @click="reconnect">Try Reconnecting</a>?
```
Comment (thebkr7) [inline src/components/common/DisconnectedBar.vue]: Nice 👍

**Pull request merged**